internal/storage/blob/store.go

```diff
@@ -214,70 +232,67 @@ func (s *Store) init(ctx context.Context) error {
 	return nil
 }
 
-func (s *Store) clone(ctx context.Context) (*CloneResult, error) {
-	ctx, cancelFunc := s.conf.getCloneCtx(ctx)
-	defer cancelFunc()
-
-	return s.cloner.Clone(ctx)
-}
-
 func (s *Store) updateIndex(ctx context.Context) error {
 	s.log.Debug("Checking for updates")
-
-	changes, err := s.clone(ctx)
+	cr, err := s.clone(ctx)
 	if err != nil {
-		return err
-	}
-
-	if failures := changes.failures(); failures > 0 {
-		s.log.Warnf("Failed to download (%d) files", failures)
+		return fmt.Errorf("failed to clone blob store: %w", err)
 	}
 
-	if changes.isEmpty() {
+	if cr.isEmpty() {
 		s.log.Debug("No changes")
 		return nil
 	}
 
-	s.log.Infof("Detected changes: added or updated (%d), deleted (%d)", len(changes.updateOrAdd), len(changes.delete))
+	s.log.Infof("Detected changes: added or updated (%d), deleted (%d)", len(cr.addedOrUpdated), len(cr.deleted))
+	dotCacheDir := filepath.Join(s.conf.WorkDir, dotcache)
```

Comment:
You can probably store this as a field in the struct because it's used in many places.